apps/molgenis-components/src/components/utils.ts

```diff
@@ -61,6 +62,52 @@ export function getPrimaryKey(
   }
 }
 
+export async function convertRowToPrimaryKey(
+  row: IRow,
+  tableName: string,
+  schemaName: string
+): Promise<Record<string, any>> {
+  const client = Client.newClient(schemaName);
+  const tableMetadata = await client.fetchTableMetaData(tableName);
+  if (!tableMetadata?.columns) {
+    throw new Error("Empty columns in metadata");
```

Comment:
Missing test for this if. Every if and else branch should be tested.

Reply:
Added the sad 😢 flow